addons/io_hubs_addon/components/operators.py

```diff
@@ -670,6 +671,212 @@ def invoke(self, context, event):
+
+            # The layer for the lightmap needs to be the active one before lightmap packing
+            obj_uv_layers.active = obj_uv_layers[LIGHTMAP_LAYER_NAME]
+            # Set the object as selected in object mode
+            obj.select_set(True)
+
+        # run UV lightmap packing on all selected objects
+        bpy.ops.object.mode_set(mode='EDIT')
+        bpy.ops.mesh.select_all(action='SELECT')
+        # bpy.ops.uv.lightmap_pack()
+        bpy.ops.uv.smart_project(island_margin=LIGHTMAP_UV_ISLAND_MARGIN)
+        bpy.ops.object.mode_set(mode='OBJECT')
+        # Deselect the objects again to return without changing the scene
+        for obj in mesh_objs:
+            obj.select_set(False)
+        # Update the view layer so all parts take notice of the changed UV layout
+        bpy.context.view_layer.update()
+
+        return {'FINISHED'}
+
+    def execute(self, context):
+        # Check selected objects
+        selected_objects = bpy.context.selected_objects
+
+        # filter mesh objects and others
+        mesh_objs, other_objs = [], []
+        for ob in selected_objects:
+            (mesh_objs if ob.type == 'MESH' else other_objs).append(ob)
+            # Remove all objects from selection so we can easily re-select subgroups later
+            ob.select_set(False)
+
+        # for ob in other_objs:
+            # Remove non-mesh objects from selection to ensure baking will work
+            # ob.select_set(False)
+
+        # Gather all materials on the selected objects
+        # materials = []
+        # Dictionary that stores which object has which materials so we can group them later
+        material_object_associations = {}
+        # Iterate over a copy of mesh_objs because we are modifying it further down
+        for obj in list(mesh_objs):
+            if len(obj.material_slots) >= 1:
+                for slot in obj.material_slots:
+                    if slot.material is not None:
+                        mat = slot.material
+                        if mat not in material_object_associations:
+                            material_object_associations[mat] = []
+                        material_object_associations[mat].append(obj)
+            else:
+                print("Object " + obj.name + " does not have material slots, removing from list of objects that will be unwrapped.")
+                mesh_objs.remove(obj)
+
+        # Set up the UV layer structure and auto-unwrap optimized for lightmaps
+        visited_materials = set()
+        for mat, obj_list in material_object_associations.items():
+            objs_to_uv_unwrap = []
+            if mat not in visited_materials:
+                # Several objects can share the same material so bundle them all
+                for ob in obj_list:
+                    for slot in ob.material_slots:
+                        if slot.material is not None:
+                            objs_to_uv_unwrap.extend(material_object_associations[slot.material])
```

Comment:
If I'm not mistaken, you may end up having an object several times in `objs_to_uv_unwrap`. You probably want to use a set and use `objs_to_uv_unwrap.update` here.